Fails during test phase for me:

```
pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform darwin -- Python 3.8.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-notify-py-0.3.3.drv-0/source
collected 20 items

tests/test_notification.py ..........F.........                          [100%]

=================================== FAILURES ===================================
______________________________ test_custom_audio _______________________________

    def test_custom_audio():
        n = notifypy.Notify(enable_logging=True)
        n.audio = "notifypy/example_notification_sound.wav"
>       assert n.send() == True
E       assert False == True
E        +  where False = <bound method Notify.send of <notifypy.notify.Notify object at 0x10c97d490>>()
E        +    where <bound method Notify.send of <notifypy.notify.Notify object at 0x10c97d490>> = <notifypy.notify.Notify object at 0x10c97d490>.send

tests/test_notification.py:79: AssertionError
----------------------------- Captured stderr call -----------------------------
2021-06-11 23:52:35.584 | INFO     | notifypy.notify:__init__:49 - Logging is enabled.
2021-06-11 23:52:35.588 | ERROR    | notifypy.os_notifiers.macos:_find_installed_afplay:84 - Unhandled exception for finding aplay.
Traceback (most recent call last):

  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
           │         │     └ {'__name__': '__main__', '__doc__': 'The pytest entry point.', '__package__': 'pytest', '__loader__': <_frozen_importlib_exte...
           │         └ <code object <module> at 0x10c5c0c90, file "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/...
           └ <function _run_code at 0x10bca5a60>
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
         │     └ {'__name__': '__main__', '__doc__': 'The pytest entry point.', '__package__': 'pytest', '__loader__': <_frozen_importlib_exte...
         └ <code object <module> at 0x10c5c0c90, file "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/...

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/pytest/__main__.py", line 5, in <module>
    raise SystemExit(pytest.console_main())
                     │      └ <function console_main at 0x10c120940>
                     └ <module 'pytest' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/pytest/...

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/config/__init__.py", line 185, in console_main
    code = main()
           └ <function main at 0x10c1208b0>

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/config/__init__.py", line 162, in main
    ret: Union[ExitCode, int] = config.hook.pytest_cmdline_main(
         │     │                │      │    └ <_HookCaller 'pytest_cmdline_main'>
         │     │                │      └ <pluggy.hooks._HookRelay object at 0x10bb55b80>
         │     │                └ <_pytest.config.Config object at 0x10c0a9f10>
         │     └ <enum 'ExitCode'>
         └ typing.Union

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'config': <_pytest.config.Config object at 0x10c0a9f10>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x10c05b1f0>
           │    │         │     └ <_HookCaller 'pytest_cmdline_main'>
           │    │         └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x10ba7b070>>
           └ <_HookCaller 'pytest_cmdline_main'>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'config': <_pytest.config.Config object at 0x10c0a9f10>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pyte...
           │    │               └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x10ba1fd30>
           └ <_pytest.config.PytestPluginManager object at 0x10ba7b070>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
                                  │     │        │       │    └ <function _multicall at 0x10c0595e0>
                                  │     │        │       └ <_HookCaller 'pytest_cmdline_main'>
                                  │     │        └ {'config': <_pytest.config.Config object at 0x10c0a9f10>}
                                  │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pyte...
                                  └ <_HookCaller 'pytest_cmdline_main'>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<_pytest.config.Config object at 0x10c0a9f10>]
          │         └ <function pytest_cmdline_main at 0x10bbdeb80>
          └ <HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytes...

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/main.py", line 316, in pytest_cmdline_main
    return wrap_session(config, _main)
           │            │       └ <function _main at 0x10bbdec10>
           │            └ <_pytest.config.Config object at 0x10c0a9f10>
           └ <function wrap_session at 0x10bbdeaf0>

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/main.py", line 269, in wrap_session
    session.exitstatus = doit(config, session) or 0
    │       │            │    │       └ <Session source exitstatus=<ExitCode.OK: 0> testsfailed=0 testscollected=20>
    │       │            │    └ <_pytest.config.Config object at 0x10c0a9f10>
    │       │            └ <function _main at 0x10bbdec10>
    │       └ <ExitCode.OK: 0>
    └ <Session source exitstatus=<ExitCode.OK: 0> testsfailed=0 testscollected=20>

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/main.py", line 323, in _main
    config.hook.pytest_runtestloop(session=session)
    │      │    │                          └ <Session source exitstatus=<ExitCode.OK: 0> testsfailed=0 testscollected=20>
    │      │    └ <_HookCaller 'pytest_runtestloop'>
    │      └ <pluggy.hooks._HookRelay object at 0x10bb55b80>
    └ <_pytest.config.Config object at 0x10c0a9f10>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'session': <Session source exitstatus=<ExitCode.OK: 0> testsfailed=0 testscollected=20>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x10c05b1f0>
           │    │         │     └ <_HookCaller 'pytest_runtestloop'>
           │    │         └ <_HookCaller 'pytest_runtestloop'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x10ba7b070>>
           └ <_HookCaller 'pytest_runtestloop'>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'session': <Session source exitstatus=<ExitCode.OK: 0> testsfailed=0 testscollected=20>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pyte...
           │    │               └ <_HookCaller 'pytest_runtestloop'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x10ba1fd30>
           └ <_pytest.config.PytestPluginManager object at 0x10ba7b070>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
                                  │     │        │       │    └ <function _multicall at 0x10c0595e0>
                                  │     │        │       └ <_HookCaller 'pytest_runtestloop'>
                                  │     │        └ {'session': <Session source exitstatus=<ExitCode.OK: 0> testsfailed=0 testscollected=20>}
                                  │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pyte...
                                  └ <_HookCaller 'pytest_runtestloop'>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Session source exitstatus=<ExitCode.OK: 0> testsfailed=0 testscollected=20>]
          │         └ <function pytest_runtestloop at 0x10bbded30>
          └ <HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytes...

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/main.py", line 348, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
    │    │                                        │              └ <Function test_custom_audio_no_file>
    │    │                                        └ <Function test_custom_audio>
    │    └ <member 'config' of 'Node' objects>
    └ <Function test_custom_audio>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_custom_audio>, 'nextitem': <Function test_custom_audio_no_file>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x10c05b1f0>
           │    │         │     └ <_HookCaller 'pytest_runtest_protocol'>
           │    │         └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x10ba7b070>>
           └ <_HookCaller 'pytest_runtest_protocol'>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_custom_audio>, 'nextitem': <Function test_custom_audio_no_file>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-...
           │    │               └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x10ba1fd30>
           └ <_pytest.config.PytestPluginManager object at 0x10ba7b070>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
                                  │     │        │       │    └ <function _multicall at 0x10c0595e0>
                                  │     │        │       └ <_HookCaller 'pytest_runtest_protocol'>
                                  │     │        └ {'item': <Function test_custom_audio>, 'nextitem': <Function test_custom_audio_no_file>}
                                  │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-...
                                  └ <_HookCaller 'pytest_runtest_protocol'>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_custom_audio>, <Function test_custom_audio_no_file>]
          │         └ <function pytest_runtest_protocol at 0x10bbd73a0>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-p...

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/runner.py", line 109, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
    │               │              └ <Function test_custom_audio_no_file>
    │               └ <Function test_custom_audio>
    └ <function runtestprotocol at 0x10bbd7430>

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/runner.py", line 126, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
    │       │      │               │             └ True
    │       │      │               └ <Function test_custom_audio>
    │       │      └ <function call_and_report at 0x10bbd7820>
    │       └ <method 'append' of 'list' objects>
    └ [<TestReport 'tests/test_notification.py::test_custom_audio' when='setup' outcome='passed'>]

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/runner.py", line 215, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
           │                 │     │       └ {}
           │                 │     └ 'call'
           │                 └ <Function test_custom_audio>
           └ <function call_runtest_hook at 0x10bbd7940>

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/runner.py", line 254, in call_runtest_hook
    return CallInfo.from_call(
           │        └ <classmethod object at 0x10bbf4d00>
           └ <class '_pytest.runner.CallInfo'>

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/runner.py", line 311, in from_call
    result: Optional[TResult] = func()
            │        │          └ <function call_runtest_hook.<locals>.<lambda> at 0x10c16ee50>
            │        └ +TResult
            └ typing.Optional

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/runner.py", line 255, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       └ {}
            │          └ <Function test_custom_audio>
            └ <_HookCaller 'pytest_runtest_call'>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_custom_audio>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x10c05b1f0>
           │    │         │     └ <_HookCaller 'pytest_runtest_call'>
           │    │         └ <_HookCaller 'pytest_runtest_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x10ba7b070>>
           └ <_HookCaller 'pytest_runtest_call'>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_custom_audio>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-...
           │    │               └ <_HookCaller 'pytest_runtest_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x10ba1fd30>
           └ <_pytest.config.PytestPluginManager object at 0x10ba7b070>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
                                  │     │        │       │    └ <function _multicall at 0x10c0595e0>
                                  │     │        │       └ <_HookCaller 'pytest_runtest_call'>
                                  │     │        └ {'item': <Function test_custom_audio>}
                                  │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-...
                                  └ <_HookCaller 'pytest_runtest_call'>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_custom_audio>]
          │         └ <function pytest_runtest_call at 0x10bbd75e0>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-p...

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/runner.py", line 162, in pytest_runtest_call
    item.runtest()
    │    └ <function Function.runtest at 0x10c0280d0>
    └ <Function test_custom_audio>

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/python.py", line 1641, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
    │    │                                   └ <Function test_custom_audio>
    │    └ <property object at 0x10c1431d0>
    └ <Function test_custom_audio>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'pyfuncitem': <Function test_custom_audio>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x10c05b1f0>
           │    │         │     └ <_HookCaller 'pytest_pyfunc_call'>
           │    │         └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x10ba7b070>>
           └ <_HookCaller 'pytest_pyfunc_call'>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'pyfuncitem': <Function test_custom_audio>}
           │    │               │     └ [<HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-...
           │    │               └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x10ba1fd30>
           └ <_pytest.config.PytestPluginManager object at 0x10ba7b070>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
                                  │     │        │       │    └ <function _multicall at 0x10c0595e0>
                                  │     │        │       └ <_HookCaller 'pytest_pyfunc_call'>
                                  │     │        └ {'pyfuncitem': <Function test_custom_audio>}
                                  │     └ [<HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-...
                                  └ <_HookCaller 'pytest_pyfunc_call'>

  File "/nix/store/djz1y4zki5giizh5nhhgap67pljqgxkq-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_custom_audio>]
          │         └ <function pytest_pyfunc_call at 0x10c019790>
          └ <HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-p...

  File "/nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/python.py", line 183, in pytest_pyfunc_call
    result = testfunction(**testargs)
             │              └ {}
             └ <function test_custom_audio at 0x10c972940>

  File "/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-notify-py-0.3.3.drv-0/source/tests/test_notification.py", line 77, in test_custom_audio
    n = notifypy.Notify(enable_logging=True)
        │        └ <class 'notifypy.notify.Notify'>
        └ <module 'notifypy' from '/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-notify-py-0.3.3.drv-0/s...

  File "/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-notify-py-0.3.3.drv-0/source/notifypy/notify.py", line 71, in __init__
    self._notifier = self._notifier_detect(**kwargs)
    │                │    │                  └ {}
    │                │    └ <class 'notifypy.os_notifiers.macos.MacOSNotifier'>
    │                └ <notifypy.notify.Notify object at 0x10c97d490>
    └ <notifypy.notify.Notify object at 0x10c97d490>

  File "/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-notify-py-0.3.3.drv-0/source/notifypy/os_notifiers/macos.py", line 52, in __init__
    call_find_afplay = self._find_installed_afplay()
                       │    └ <staticmethod object at 0x10c879af0>
                       └ <notifypy.os_notifiers.macos.MacOSNotifier object at 0x10c97ddc0>

> File "/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-notify-py-0.3.3.drv-0/source/notifypy/os_notifiers/macos.py", line 78, in _find_installed_afplay
    run_which_for_aplay = subprocess.check_output(["which", "afplay"])
                          │          └ <function check_output at 0x10c5a6430>
                          └ <module 'subprocess' from '/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/subprocess.py'>

  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           │    │                 │             └ None
           │    │                 └ -1
           │    └ (['which', 'afplay'],)
           └ <function run at 0x10c5a64c0>
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/subprocess.py", line 493, in run
    with Popen(*popenargs, **kwargs) as process:
         │      │            └ {'stdout': -1}
         │      └ (['which', 'afplay'],)
         └ <class 'subprocess.Popen'>
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
    │    │              │     │           │           └ True
    │    │              │     │           └ None
    │    │              │     └ None
    │    │              └ ['which', 'afplay']
    │    └ <function Popen._execute_child at 0x10c5a53a0>
    └ <subprocess.Popen object at 0x10c97d5b0>
  File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
          │                    │          │        └ 'which'
          │                    │          └ 'No such file or directory'
          │                    └ 2
          └ <class 'OSError'>

FileNotFoundError: [Errno 2] No such file or directory: 'which'
2021-06-11 23:52:35.598 | WARNING  | notifypy.os_notifiers.macos:send_notification:96 - Notification icon is not supported. Read the docs for more information.
=============================== warnings summary ===============================
tests/test_notification.py::test_custom_audio
  /nix/store/r43si7s6s83iymk0vr9sq64w5bq4znfd-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/threadexception.py:75: PytestUnhandledThreadExceptionWarning: Exception in thread notify.py

  Traceback (most recent call last):
    File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/threading.py", line 932, in _bootstrap_inner
      self.run()
    File "/nix/store/p6k14ihyfaq3z4sycxr09cvgpis2zjd5-python3-3.8.9/lib/python3.8/threading.py", line 870, in run
      self._target(*self._args, **self._kwargs)
    File "/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-notify-py-0.3.3.drv-0/source/notifypy/notify.py", line 268, in <lambda>
      target=lambda: self.start_notification_thread(event)
    File "/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-notify-py-0.3.3.drv-0/source/notifypy/notify.py", line 287, in start_notification_thread
      result = self.send_notification(
    File "/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-notify-py-0.3.3.drv-0/source/notifypy/notify.py", line 323, in send_notification
      attempt_to_send_notifiation = self._notifier.send_notification(
    File "/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-notify-py-0.3.3.drv-0/source/notifypy/os_notifiers/macos.py", line 110, in send_notification
      raise BinaryNotFound("afplay")
  notifypy.exceptions.BinaryNotFound: Unable to find required afplay. Please check if it's installed.

    warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED tests/test_notification.py::test_custom_audio - assert False == True
=================== 1 failed, 19 passed, 1 warning in 3.47s ====================
builder for '/nix/store/hg3qd76wxyafwyd4vrhvgx99nw0pqq27-python3.8-notify-py-0.3.3.drv' failed with exit code 1
error: build of '/nix/store/hg3qd76wxyafwyd4vrhvgx99nw0pqq27-python3.8-notify-py-0.3.3.drv' failed
```

--- a/pkgs/development/python-modules/notify-py/darwin-paths.patch
+++ b/pkgs/development/python-modules/notify-py/darwin-paths.patch
@@ -1,0 +1,13 @@
+diff --git a/notifypy/os_notifiers/macos.py b/notifypy/os_notifiers/macos.py
+index 68731fb..53fcee3 100644
+--- a/notifypy/os_notifiers/macos.py
++++ b/notifypy/os_notifiers/macos.py
+@@ -75,7 +75,7 @@ class MacOSNotifier(BaseNotifier):
+     def _find_installed_afplay():
+         """Function to find the path for afplay"""
+         try:
+-            run_which_for_aplay = subprocess.check_output(["which", "afplay"])
++            run_which_for_aplay = subprocess.check_output(["@which@", "afplay"])
+             return run_which_for_aplay.decode("utf-8")
+         except subprocess.CalledProcessError:
+             logger.exception("Unable to find aplay.")

--- a/pkgs/development/python-modules/notify-py/default.nix
+++ b/pkgs/development/python-modules/notify-py/default.nix
@@ -1,5 +1,17 @@
-{ lib, stdenv, buildPythonPackage, fetchFromGitHub, isPy3k, coreutils, alsa-utils
-, libnotify, which, jeepney, loguru, pytestCheckHook }:
+{ lib
+, stdenv
+, buildPythonPackage
+, isPy3k
+, fetchFromGitHub
+, substituteAll
+, alsa-utils
+, libnotify
+, which
+, jeepney
+, loguru
+, pytestCheckHook
+, coreutils
+}:
 
 buildPythonPackage rec {
   pname = "notify-py";
@@ -14,15 +26,28 @@ buildPythonPackage rec {
     sha256 = "1n35adwsyhz304n4ifnsz6qzkymwhyqc8sg8d76qv5psv2xsnzlf";
   };
 
-  propagatedNativeBuildInputs = [ which ]
-    ++ lib.optionals stdenv.isLinux [ alsa-utils libnotify ];
+  patches = lib.optionals stdenv.isLinux [
+    # hardcode paths to aplay and notify-send
+    (substituteAll {
+      src = ./linux-paths.patch;
+      aplay = "${alsa-utils}/bin/aplay";
+      notifysend = "${libnotify}/bin/notify-send";
+    })
+  ] ++ lib.optionals stdenv.isDarwin [
+    # hardcode path to which
+    (substituteAll {
+      src = ./darwin-paths.patch;
+      which = "${which}/bin/which";
+    })
+  ];
+
   propagatedBuildInputs = [ loguru ]
     ++ lib.optionals stdenv.isLinux [ jeepney ];
 
-  checkInputs = [ coreutils pytestCheckHook ];
+  checkInputs = [ pytestCheckHook ];
 
   # Tests search for "afplay" binary which is built in to MacOS and not available in nixpkgs
-  preCheck = ''
+  preCheck = lib.optionalString stdenv.isDarwin ''
     mkdir $TMP/bin
     ln -s ${coreutils}/bin/true $TMP/bin/afplay
     export PATH="$TMP/bin:$PATH"
@@ -31,9 +56,9 @@ buildPythonPackage rec {
   pythonImportsCheck = [ "notifypy" ];
 
   meta = with lib; {
-    description = "Python Module for sending cross-platform desktop notifications on Windows, macOS, and Linux.";
-    homepage = "https://github.com/ms7m/notify-py/";
+    description = "Cross-platform desktop notification library for Python";
+    homepage = "https://github.com/ms7m/notify-py";
     license = licenses.mit;
-    maintainers = with maintainers; [ austinbutler ];
+    maintainers = with maintainers; [ austinbutler dotlambda ];
   };
 }

--- a/pkgs/development/python-modules/notify-py/linux-paths.patch
+++ b/pkgs/development/python-modules/notify-py/linux-paths.patch
@@ -1,0 +1,54 @@
+diff --git a/notifypy/os_notifiers/linux.py b/notifypy/os_notifiers/linux.py
+index ee89216..5201574 100644
+--- a/notifypy/os_notifiers/linux.py
++++ b/notifypy/os_notifiers/linux.py
+@@ -53,30 +53,12 @@ class LinuxNotifierLibNotify(BaseNotifier):
+     @staticmethod
+     def _find_installed_aplay():
+         """Function to find the path for notify-send"""
+-        try:
+-            run_which_for_aplay = subprocess.check_output(["which", "aplay"])
+-            return run_which_for_aplay.decode("utf-8")
+-        except subprocess.CalledProcessError:
+-            logger.exception("Unable to find aplay.")
+-            return False
+-        except Exception:
+-            logger.exception("Unhandled exception for finding aplay.")
+-            return False
++        return "@aplay@"
+ 
+     @staticmethod
+     def _find_installed_notify_send():
+         """Function to find the path for notify-send"""
+-        try:
+-            run_which_for_notify_send = subprocess.check_output(
+-                ["which", "notify-send"]
+-            )
+-            return run_which_for_notify_send.decode("utf-8")
+-        except subprocess.CalledProcessError:
+-            logger.exception("Unable to find notify-send.")
+-            return False
+-        except Exception:
+-            logger.exception("Unhandled exception for finding notify-send.")
+-            return False
++        return "@notifysend@"
+ 
+     def send_notification(
+         self,
+@@ -159,15 +141,7 @@ class LinuxNotifier(BaseNotifier):
+     @staticmethod
+     def _find_installed_aplay():
+         """Function to find the path for notify-send"""
+-        try:
+-            run_which_for_aplay = subprocess.check_output(["which", "aplay"])
+-            return run_which_for_aplay.decode("utf-8")
+-        except subprocess.CalledProcessError:
+-            logger.exception("Unable to find aplay.")
+-            return False
+-        except Exception:
+-            logger.exception("Unhandled exception for finding aplay.")
+-            return False
++        return "@aplay@"
+ 
+     def send_notification(
+         self,

